api/base/parsers.py

```diff
@@ -42,44 +45,78 @@ def flatten_relationships(self, relationships):
         id = data.get('id')
         return {'id': id, 'target_type': target_type}
 
+    def flatten_data(self, resource_object, request_method, is_list, is_relationship):
```

Comment:
Pretty minor: The signature of this method can be simplified, since `request_method`, `is_list`, and `is_relationship` can all be derived from `parser_context`.

```python
def flatten_data(self, resource_object, parser_context):
    request = parser_context['request']
    is_relationship = parser_context.get('is_relationship')
    #...

def parse(...):
    #...
    self.flatten_data(data_object, parser_context)
```